DataMigration/src/main/java/org/sagebionetoworks/dian/datamigration/SynapseUtil.java

```diff
@@ -0,0 +1,970 @@
+        }
+
+        // Download all test session schedule files
+        File testSessionScheduleFolder = createFolderIfNecessary(SESSION_SCHEDULE_FOLDER_NAME);
+        File testSessionScheduleExtractedFolder = createFolderIfNecessary(
+                SESSION_SCHEDULE_FOLDER_NAME + EXTRACTED_SUFFIX);
+        if (testSessionScheduleExtractedFolder != null && testSessionScheduleFolder != null) {
+            for (FileHandleAssociation file : files.sessionSchedules) {
+                File download = new File(SESSION_SCHEDULE_FOLDER_NAME +
+                        File.separator + file.getAssociateObjectId() + ZIP);
+                System.out.println("Downloading file " + file.getAssociateObjectId() + ".zip");
+                synapse.downloadFile(file, download);
+                System.out.println("Unzipping file " + file.getAssociateObjectId() + ".zip");
+                UnzipUtil.unzip(download.getAbsolutePath(),
+                        testSessionScheduleExtractedFolder.getAbsolutePath());
+            }
+        }
+
+        // Download all wake sleep schedule files
+        File wakeSleepScheduleFolder = createFolderIfNecessary(WAKE_SLEEP_FOLDER_NAME);
+        File wakeSleepScheduleExtractedFolder = createFolderIfNecessary(
+                WAKE_SLEEP_FOLDER_NAME + EXTRACTED_SUFFIX);
+        if (wakeSleepScheduleExtractedFolder != null && wakeSleepScheduleFolder != null) {
+            for (FileHandleAssociation file : files.wakeSleepSchedules) {
+                File download = new File(WAKE_SLEEP_FOLDER_NAME +
+                        File.separator + file.getAssociateObjectId() + ZIP);
+                System.out.println("Downloading file " + file.getAssociateObjectId() + ".zip");
+                synapse.downloadFile(file, download);
+                System.out.println("Unzipping file " + file.getAssociateObjectId() + ".zip");
+                UnzipUtil.unzip(download.getAbsolutePath(),
+                        wakeSleepScheduleExtractedFolder.getAbsolutePath());
+            }
+        }
+
+        DianFileFolders folders = new DianFileFolders();
+        folders.testSessions = testSessionExtractedFolder;
+        folders.sessionSchedules = testSessionScheduleExtractedFolder;
+        folders.wakeSleepSchedules = wakeSleepScheduleExtractedFolder;
+        return folders;
+    }
+
+    /**
+     * @param files to download
+     * @throws SynapseException throws if download fails
+     */
+    public static DianParticipantFileFolders downloadDianParticipantFiles(
+            DianParticipantFiles files) throws SynapseException, IOException {
+
+        // Download and unzip the hasd and exr participants JSON
+        File hasdParticipantsFolder = createFolderIfNecessary(HASD_PARTICIPANTS_FOLDER_NAME);
+        File hasdParticipantsExtractedFolder = createFolderIfNecessary(
+                HASD_PARTICIPANTS_FOLDER_NAME + EXTRACTED_SUFFIX);
+        if (hasdParticipantsFolder != null && hasdParticipantsExtractedFolder != null) {
```

Comment:
I had similar thoughts while writing the code that I wasn't properly throwing exceptions.  I'm so used to writing crash-proof java code for Android, that this is a bit of a paradigm shift for me. I'll switch everything to throw exceptions, I see many places where this can be improved from your PR comments.